### ISSUE TITLE
Replace use of deprecated function Base.download with something that's not deprecated

### ISSUE THREAD

**Issue by BoZenKhaa** [BoZenKhaa/ReverseGeocode.jl#21]
It was deprecated in v1.6 https://docs.julialang.org/en/v1.6/NEWS/

**Comment by pywugate** [comment]
I think I add `Downloads` into dependency. 


```julia
(@v1.10) pkg> dev ReverseGeocode

(@v1.10) pkg> activate ReverseGeocode
  Activating project at `~\.julia\dev\ReverseGeocode`

(ReverseGeocode) pkg> st
Project ReverseGeocode v0.5.0
Status `~\.julia\dev\ReverseGeocode\Project.toml`
  [336ed68f] CSV v0.10.14
  [a93c6f00] DataFrames v1.6.1
  [864edb3b] DataStructures v0.18.20
  [b8a86587] NearestNeighbors v0.4.16
  [a5390f91] ZipFile v0.10.1
  [f43a241f] Downloads v1.6.0
  [56ddb016] Logging

(ReverseGeocode) pkg> add Downloads
   Resolving package versions...
    Updating `~\.julia\dev\ReverseGeocode\Project.toml`
  [f43a241f] + Downloads v1.6.0
    Updating `~\.julia\dev\ReverseGeocode\Manifest.toml`
  [0dad84c5] + ArgTools v1.1.1
  [f43a241f] + Downloads v1.6.0
  [7b1f6079] + FileWatching
  [b27032c2] + LibCURL v0.6.4
  [ca575930] + NetworkOptions v1.2.0
  [deac9b47] + LibCURL_jll v8.4.0+0
  [29816b5a] + LibSSH2_jll v1.11.0+1
  [c8ffd9c3] + MbedTLS_jll v2.28.2+1
  [14a3606d] + MozillaCACerts_jll v2023.1.10
  [8e850ede] + nghttp2_jll v1.52.0+1

(ReverseGeocode) pkg> instantiate
(ReverseGeocode) pkg> update
    Updating registry at `~\.julia\registries\General`
    Updating git-repo `https://github.com/JuliaRegistries/General.git`
    Updating registry at `~\.julia\registries\HolyLabRegistry`
    Updating git-repo `git@github.com:HolyLab/HolyLabRegistry.git`
  No Changes to `~\.julia\dev\ReverseGeocode\Project.toml`
  No Changes to `~\.julia\dev\ReverseGeocode\Manifest.toml`

``` 

the `project.toml` have `Downloads` in [deps] but not in [compat]

```
name = "ReverseGeocode"
uuid = "96b87e3f-b46e-456c-8178-0e438baab6fa"
authors = ["Jan Mrkos <mrkosjan@gmail.com>"]
version = "0.5.0"

[deps]
CSV = "336ed68f-0bac-5ca0-87d4-7b16caf5d00b"
DataFrames = "a93c6f00-e57d-5684-b7b6-d8193f3e46c0"
DataStructures = "864edb3b-99cc-5e75-8d2d-829cb0a9cfe8"
Downloads = "f43a241f-c20a-4ad4-852c-f6b1247861c6"
Logging = "56ddb016-857b-54e1-b83d-db4d58db5568"
NearestNeighbors = "b8a86587-4115-5ab1-83bc-aa920d37bbce"
ZipFile = "a5390f91-8eb1-5f08-bee0-b1d1ffed6cea"

[compat]
CSV = "0.10"
DataFrames = "1"
DataStructures = "0.18"
NearestNeighbors = "0.4"
ZipFile = "0.10"
julia = "1"

```

```julia 
julia> using ReverseGeocode
julia> Geocoder();
[ Info: Downloading 'countryInfo.txt' from http://download.geonames.org/export/dump/countryInfo.txt.
[ Info: cities1000.csv not found in ~\.julia\dev\ReverseGeocode\data.
[ Info: Downloading cities1000 zip from http://download.geonames.org/export/dump/cities1000.zip.
[ Info: Processing cities1000.zip.
[ Info: Reference dataset sucessfuly extracted and saved in ~\.julia\dev\ReverseGeocode\data\cities1000.csv.
```

**Comment by pywugate** [comment]
But I don't know how to create a new pull request without my previous pull request.
(This is my first time to contribute existing package with other on Github.)

**Comment by BoZenKhaa** [comment]
> I think I add Downloads into dependency.

Ok, so I just looked at the `download` function, and `Downloads` package is part of Julia standard library, so you don't need to install it. Just use 
```Julia
using Downloads

Downloads.download(......
```
That way, `project.toml` does not need to be updated.

> I don't know how to create a new pull request without my previous pull request.

No worries, you can open a PR from any branch of your forked repository. So, to make a new PR, start a new branch from the master (that corresponds to the latest release here) and put your changes there. Once you push that branch into your fork, you can open a PR with just the changes to the download.

**Comment by pywugate** [comment]
I tried directly add `Downloads` in `ReverseGeocode.jl`
but it failed 
```julia
using ZipFile,
    CSV,
    NearestNeighbors,
    Logging,
    DataFrames,
    DataStructures,
    Downloads
```

```julia 
julia> using ReverseGeocode

Precompiling ReverseGeocode
        Info Given ReverseGeocode was explicitly requested, output will be shown live
ERROR: LoadError: ArgumentError: Package ReverseGeocode does not have Downloads in its dependencies:
- You may have a partially installed environment. Try `Pkg.instantiate()`
  to ensure all packages in the environment are installed.
- Or, if you have ReverseGeocode checked out for development and have
  added Downloads as a dependency but haven't updated your primary
  environment's manifest file, try `Pkg.resolve()`.
- Otherwise you may need to report an issue with ReverseGeocode
Stacktrace:
 [1] macro expansion
   @ .\loading.jl:1776 [inlined]
 [2] macro expansion
   @ .\lock.jl:267 [inlined]
 [3] __require(into::Module, mod::Symbol)
   @ Base .\loading.jl:1753
 [4] #invoke_in_world#3
   @ .\essentials.jl:926 [inlined]
 [5] invoke_in_world
   @ .\essentials.jl:923 [inlined]
 [6] require(into::Module, mod::Symbol)
   @ Base .\loading.jl:1746
 [7] include
   @ .\Base.jl:495 [inlined]
 [8] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, 
load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt128}}, source::Nothing)
   @ Base .\loading.jl:2222
 [9] top-level scope
   @ stdin:3
in expression starting at ~\.julia\packages\ReverseGeocode\kue0J\src\ReverseGeocode.jl:1
in expression starting at stdin:3
  ✗ ReverseGeocode
  0 dependencies successfully precompiled in 3 seconds. 48 already precompiled.

ERROR: The following 1 direct dependency failed to precompile:

ReverseGeocode [96b87e3f-b46e-456c-8178-0e438baab6fa]

Failed to precompile ReverseGeocode [96b87e3f-b46e-456c-8178-0e438baab6fa] to "~\\.julia\\compiled\\v1.10\\ReverseGeocode\\jl_AC6B.tmp".
ERROR: LoadError: ArgumentError: Package ReverseGeocode does not have Downloads in its dependencies:
- You may have a partially installed environment. Try `Pkg.instantiate()`
  to ensure all packages in the environment are installed.
- Or, if you have ReverseGeocode checked out for development and have
  added Downloads as a dependency but haven't updated your primary
  environment's manifest file, try `Pkg.resolve()`.
- Otherwise you may need to report an issue with ReverseGeocode
Stacktrace:
 [1] macro expansion
   @ .\loading.jl:1776 [inlined]
 [2] macro expansion
   @ .\lock.jl:267 [inlined]
 [3] __require(into::Module, mod::Symbol)
   @ Base .\loading.jl:1753
 [4] #invoke_in_world#3
   @ .\essentials.jl:926 [inlined]
 [5] invoke_in_world
   @ .\essentials.jl:923 [inlined]
 [6] require(into::Module, mod::Symbol)
   @ Base .\loading.jl:1746
 [7] include
   @ .\Base.jl:495 [inlined]
 [8] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, 
load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt128}}, source::Nothing)
   @ Base .\loading.jl:2222
 [9] top-level scope
   @ stdin:3
in expression starting at ~\.julia\packages\ReverseGeocode\kue0J\src\ReverseGeocode.jl:1
in expression starting at stdin:
Stacktrace:
  [1] pkgerror(msg::String)
    @ Pkg.Types ~\AppData\Local\Programs\Julia-1.10.2\share\julia\stdlib\v1.10\Pkg\src\Types.jl:70     
  [2] precompile(ctx::Pkg.Types.Context, pkgs::Vector{…}; internal_call::Bool, strict::Bool, warn_loaded::Bool, already_instantiated::Bool, timing::Bool, _from_loading::Bool, kwargs::@Kwargs{…})
    @ Pkg.API ~\AppData\Local\Programs\Julia-1.10.2\share\julia\stdlib\v1.10\Pkg\src\API.jl:1659       
  [3] precompile(pkgs::Vector{Pkg.Types.PackageSpec}; io::Base.TTY, kwargs::@Kwargs{_from_loading::Bool})
    @ Pkg.API ~\AppData\Local\Programs\Julia-1.10.2\share\julia\stdlib\v1.10\Pkg\src\API.jl:159        
  [4] precompile
    @ ~\AppData\Local\Programs\Julia-1.10.2\share\julia\stdlib\v1.10\Pkg\src\API.jl:147 [inlined]      
  [5] #precompile#114
    @ ~\AppData\Local\Programs\Julia-1.10.2\share\julia\stdlib\v1.10\Pkg\src\API.jl:146 [inlined]      
  [6] #invokelatest#2
    @ .\essentials.jl:894 [inlined]
  [7] invokelatest
    @ .\essentials.jl:889 [inlined]
  [8] _require(pkg::Base.PkgId, env::String)
    @ Base .\loading.jl:1963
  [9] __require_prelocked(uuidkey::Base.PkgId, env::String)
    @ Base .\loading.jl:1812
 [10] #invoke_in_world#3
    @ .\essentials.jl:926 [inlined]
 [11] invoke_in_world
    @ .\essentials.jl:923 [inlined]
 [12] _require_prelocked(uuidkey::Base.PkgId, env::String)
    @ Base .\loading.jl:1803
 [13] macro expansion
    @ .\loading.jl:1790 [inlined]
 [14] macro expansion
    @ .\lock.jl:267 [inlined]
 [15] __require(into::Module, mod::Symbol)
    @ Base .\loading.jl:1753
 [16] #invoke_in_world#3
    @ .\essentials.jl:926 [inlined]
 [17] invoke_in_world
    @ .\essentials.jl:923 [inlined]
 [18] require(into::Module, mod::Symbol)
    @ Base .\loading.jl:1746
Some type information was truncated. Use `show(err)` to see complete types.
```

**Comment by BoZenKhaa** [comment]
You are right, I just tested this, and it only works with installed `Downloads`. The changes work nicely, though, with no warnings :-). So, all good. Anyway, I think we need the compact entry for the new package version to be registered without issues. Can you put 1.5.3 as the `Downloads` version there? There is something weird going on with the versions in that package and [1.5.3. is the latest relaease.](https://github.com/JuliaLang/Downloads.jl)

**Comment by pywugate** [comment]
no matter how I specify the version of Downloads, it always gives add Downloads 1.6, 
as same as in [issue#238](https://github.com/JuliaLang/Downloads.jl/issues/238)
and it is indeed 1.6. in the [commit](https://github.com/JuliaLang/Downloads.jl/commit/43899e71eed709866f96dd624f1c369a1c8e26a3)

somehow the version number on github page isn't updated.

**Comment by BoZenKhaa** [comment]
Oh ok, thanks, the threads you link are wild. v1.6 it is!

**Comment by BoZenKhaa** [comment]
Addressed by #22 

@JuliaRegistrator register
